Source/debug.cpp

```diff
@@ -101,13 +106,41 @@ std::string DebugCmdHelp(const std::string_view parameter)
 
 std::string DebugCmdGiveGoldCheat(const std::string_view parameter)
 {
-	GiveGoldCheat();
+	auto &myPlayer = Players[MyPlayerId];
+
+	for (int8_t &itemId : myPlayer.InvGrid) {
+		if (itemId != 0)
+			continue;
+
+		int ni = myPlayer._pNumInv++;
+		SetPlrHandItem(&myPlayer.InvList[ni], IDI_GOLD);
+		GetPlrHandSeed(&myPlayer.InvList[ni]);
+		myPlayer.InvList[ni]._ivalue = GOLD_MAX_LIMIT;
+		myPlayer.InvList[ni]._iCurs = ICURS_GOLD_LARGE;
+		myPlayer._pGold += GOLD_MAX_LIMIT;
+		itemId = myPlayer._pNumInv;
+	}
+
 	return "You are now rich! If only this was as easy in real life...";
 }
 
 std::string DebugCmdTakeGoldCheat(const std::string_view parameter)
 {
-	TakeGoldCheat();
+	auto &myPlayer = Players[MyPlayerId];
+
+	for (auto itemId : myPlayer.InvGrid) {
+		itemId -= 1;

```

Comment:
how is this supposed to work?

Reply:
Idk, it was like this when I found it, just moved the code